server/pulp/server/async/worker_watcher.py

```diff
@@ -72,14 +71,14 @@ def handle_worker_heartbeat(event):
     :type event: dict
     """
     event_info = _parse_and_log_event(event)
-    worker = Worker.objects(name=event_info['worker_name']).first()
-
-    if not worker:
+    try:
```

Comment:
This looks like a good opportunity for ``get_or_create``:

https://docs.djangoproject.com/en/1.10/ref/models/querysets/#get-or-create